File: src/Helveg.Visualization/Multigraph.cs
using System;
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Helveg.Visualization;

public record Multigraph(
    string Id,
    string? Label,
    ImmutableArray<Node> Nodes,
    ImmutableArray<Relation> Relations);

File: src/UI/App/model/multigraph.ts
export interface Multigraph {
    id: string,
    nodes: Record<string, Node>
    relations: Record<string, Relation>
}

export enum DiagnosticSeverity {
    Unknown = "Unknown",
    Hidden = "Hidden",
    Info = "Info",
    Warning = "Warning",
    Error = "Error"
}

export interface Diagnostic {
    id: string,
    message: string,
    severity: DiagnosticSeverity,
}

export interface NodeProperties {
    Label?: string,
    Diagnostics?: Diagnostic[]
}

export interface Node {
    properties: NodeProperties & Record<string, any>
}

export interface Relation {
    edges: Edge[]
}

export interface EdgeProperties {
    Label?: string
}

export interface Edge {
    src: string,
    dst: string,
    properties: EdgeProperties & Record<string, any>
}
